omod/src/main/java/org/openmrs/module/datafilter/web/controller/DataFilterController.java

```diff
@@ -0,0 +1,127 @@
+import org.openmrs.module.datafilter.impl.api.DataFilterDefaultResponse;
+import org.openmrs.module.datafilter.impl.api.DataFilterService;
+import org.openmrs.module.datafilter.impl.api.EntityBasisMapResponseMapper;
+import org.openmrs.module.datafilter.impl.api.EntityBasisMapSearchRequest;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.validation.Valid;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/datafilter")
+public class DataFilterController extends BaseRestController {
+	
+	private Log log = LogFactory.getLog(this.getClass());
+	
+	@Autowired
+	private DataFilterService dataFilterService;
+	
+	@Autowired
+	private EntityBasisMapResponseMapper dataFilterMapper;
+	
+	@RequestMapping(method = RequestMethod.POST, value = "entitybasismap")
+	@ResponseBody
+	public ResponseEntity<Object> grantAccess(@Valid @RequestBody EntityBasisMap entityBasisMapRequest) {
+		try {
+			OpenmrsObject entity = null;
+			OpenmrsObject basis = null;
+			String entityDomain = entityBasisMapRequest.getEntityType();
+			String entityIdentifier = entityBasisMapRequest.getEntityIdentifier();
+			String basisDomain = entityBasisMapRequest.getBasisType();
+			String basisIdentifier = entityBasisMapRequest.getBasisIdentifier();
+			
+			if (StringUtils.isNotBlank(entityDomain) && StringUtils.isNotBlank(entityIdentifier)
+			        && entityDomain.equalsIgnoreCase("org.openmrs.User")) {
+				entity = Context.getUserService().getUserByUuid(entityIdentifier);
+			} else if (StringUtils.isNotBlank(entityDomain) && StringUtils.isNotBlank(entityIdentifier)
+			        && entityDomain.equalsIgnoreCase("org.openmrs.Patient")) {
+				entity = Context.getPatientService().getPatientByUuid(entityIdentifier);
+			}
+			
+			if (StringUtils.isNotBlank(basisDomain) && StringUtils.isNotBlank(basisIdentifier)
+			        && basisDomain.equalsIgnoreCase("org.openmrs.Location")) {
+				basis = Context.getLocationService().getLocationByUuid(basisIdentifier);
+			}
+			if (entity == null) {
+				return new ResponseEntity<>("Data filter Entity is required!", HttpStatus.BAD_REQUEST);
+			}
+			
+			if (basis == null) {
+				return new ResponseEntity<>("Data filter Basis is required!", HttpStatus.BAD_REQUEST);
+			}
+			dataFilterService.grantAccess(entity, basis);
```

Comment:
The methods like grantAccess are not protected in the API, this will be a security hole because anyone can call this endpoint to grant themselves to forbidden resources.